client/extensions/woocommerce/app/settings/payments/payments-off-site.js

```diff
@@ -58,12 +62,29 @@ class SettingsPaymentsOffSite extends Component {
 							<ListItemField className="payments__methods-column-settings">
 							</ListItemField>
 						</ListHeader>
-						{ methods && methods.map( this.renderMethodItem ) }
+						{ paymentMethods && paymentMethods.map( this.renderMethodItem ) }
 					</List>
 			</div>
 		);
 	}
+}
+
+function mapStateToProps( state ) {
+	const siteId = getSelectedSiteId( state );
+	const paymentMethods = getPaymentMethodsGroup( state, siteId, 'off-site' );
+	return {
+		paymentMethods,
+		siteId,
```

Comment:
This doesn't feel right. The UI components shouldn't know anything about `siteId`. Instead, in `mapDispatchToProps` , you should return functions that already have the `siteId` param set. What do you think about that?